### PR TITLE
[INTEL MKL] Added input name in the bfloat16 namescope interface so t…

### PR DESCRIPTION
**Pull request by cuixiaom**
…hat it can create correct node name when it is needed. The default value of the name is '', which will make sure the change is backward compatible, and won't affect existing models which suse bfloat16 namescope

### CODE REVIEW

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@cuixiaom Can you please fix build failures ? Thanks!

**Comment by cuixiaom** [comment]
> @cuixiaom Can you please fix build failures ? Thanks!

Yes, I have fixed the format which caused CI failure, the other failures were not caused by my changes.

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by mihaimaruseac** [comment]
Since this changes `defaults` from `None` to `[]` it now needs a review from API owners

File: tensorflow/python/tpu/bfloat16.py
Comment (karmel) [inline tensorflow/python/tpu/bfloat16.py:73]: Let's make the default None here, and then set to `''` if None, as per the API policy: https://github.com/tensorflow/community/blob/master/governance/api-reviews.md#optional-arguments-with-default-values

**Review by karmel — CHANGES_REQUESTED** [review]
For @tensorflow/api-owners 

Note that this will not be included in any v1 release. Is there a particular reason we want to update the v1 APIs here? 

@reedwm - is there a way to do this with v2 APIs? We want to make sure we're not locking people in to v1 APIs.

**Comment by reedwm** [comment]
> @reedwm - is there a way to do this with v2 APIs? We want to make sure we're not locking people in to v1 APIs.

The `tf.keras.mixed_precision` API is the recommended way of using bfloat16 in TF 2. Altneratively, 
this can be done with a grappler pass once I update and merge #34504 (I'll do this soon), but we strongly recommend `tf.keras.mixed_precision` over the grappler pass.

**Comment by martinwicke** [comment]
Thanks Reed. That leaves the question Karmel asked earlier.

File: tensorflow/python/tpu/bfloat16.py
Comment (cuixiaom) [reply tensorflow/python/tpu/bfloat16.py:73]: I used '' instead of "None" is because of backward compatibility. Since there are already some user use the tensorflow without my change to train their bfloat16 model, they didn't provide the bfloat16 namescope name, therefore in their trained model, the name of bfloat16 namescope is default value ''. If I change the '' into "None", they have to provide a name to their bfloat16 namescope, and that will cause the node name change in their models, and the inference will fail.  If they don't provide the name, Tensorflow will raise error like follows:
tensorflow/python/ops/variable_scope.py", line 2278, in __init__
    raise TypeError("If default_name is None then name_or_scope is required")

File: tensorflow/tools/api/golden/v1/tensorflow.tpu.pbtxt
Comment (mihaimaruseac) [inline tensorflow/tools/api/golden/v1/tensorflow.tpu.pbtxt:21]: Does the golden need to be regenerated?

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/tools/api/golden/v1/tensorflow.tpu.pbtxt
Comment (cuixiaom) [reply tensorflow/tools/api/golden/v1/tensorflow.tpu.pbtxt]: yes, I have updated the golden. 
thanks!

**Review by mihaimaruseac — APPROVED** [review]

**Comment by jsimsa** [comment]
For @tensorflow/api-owners:

@cuixiaom Note that this will not be included in any v1 release. Is there a particular reason we want to update the v1 APIs here?

**Pull request merged**
